tools/generator/GenBaseSupport.cs

```diff
@@ -302,12 +302,14 @@ public override bool ValidateNamespace ()
 			return true;
 		}
 
-		bool IsObfuscatedName (String name)
+		bool IsObfuscatedName (int threshold, string name)
 		{
 			if (name.StartsWith ("R.", StringComparison.Ordinal))
 				return false;
 			int idx = name.LastIndexOf ('.');
 			string last = idx < 0 ? name : name.Substring (idx + 1);
+			if (threshold <= Math.Pow (26, last.Length - 1))
```

Comment:
Seealso: "id" in real world use. https://developer.android.com/reference/android/R.id.html